app/presenters/configuration_status_presenter.rb

```diff
@@ -0,0 +1,254 @@
+  end
+
+  def total_data_query
+    host_status_table
+      .where(host_status_table[:type].eq('HostStatus::ConfigurationStatus')
+        .and(is_relevant))
+      .join(hosts_table)
+        .on(host_status_table[:host_id].eq(hosts_table[:id])
+          .and(hosts_table[:type].eq('Host::Managed')))
+      .project(computed_status)
+      .project(arel_quoted(1).count)
+      .group('computed_status')
+  end
+
+  def owned_data_query
+    host_status_table
+      .where(host_status_table[:type].eq('HostStatus::ConfigurationStatus')
+        .and(
+          host_status_table[:host_id].in(
+            Arel.sql(Host::Managed.search_for('owner = current_user').select(:id).to_sql)
+          )
+        )
+        .and(is_relevant))
+      .join(hosts_table).on(host_status_table[:host_id].eq(hosts_table[:id]).and(hosts_table[:type].eq('Host::Managed')))
+      .project(computed_status)
+      .project(arel_quoted(1).count)
+      .group('computed_status')
+  end
+
+  def is_relevant
+    hosts_table[:puppet_proxy_id].eq(nil).not
+      .or(any_reports)
+      .or(always_show_configuration_status)
+  end
+
+  def always_show_configuration_status
+    settings_table
+      .where(settings_table[:name].eq('always_show_configuration_status')
+        .and(settings_table[:value].eq('--- true')
+          .or(settings_table[:value].eq(nil).and(settings_table[:default].eq('--- true')))))
+      .exists
+  end
+
+  # 0000 0000000000 0000000000 0000000000 0000000000 0000000000 0000000000
+  # ----     [6]        [5]        [4]        [3]        [2]        [1]
+  # [1] - applied
+  # [2] - restarted
+  # [3] - failed
+  # [4] - failed_restarts
+  # [5] - skipped
+  # [6] - pending
+
+  # WHEN pending THEN pending
+  # WHEN failed OR failed_restarts THEN error
+  # WHEN applied OR restarted THEN active
+  # ELSE no_changes
+
+  def computed_status
+    arel_case
+      .when(alerts_disabled).then(status_class::ALERTS_DISABLED)
+      .when(no_reports).then(status_class::NO_REPORTS)
+      .when(out_of_sync).then(status_class::OUT_OF_SYNC)
+      .when(pending).then(status_class::PENDING)
+      .when(failed_or_failed_restarts).then(status_class::ERROR)
+      .when(applied_or_restarted).then(status_class::ACTIVE)
+      .else(status_class::NO_CHANGES)
+      .as('computed_status')
+  end
+
+  def alerts_disabled
+    hosts_table[:enabled].eq(false)
+  end
+
+  def any_reports
+    reports_table
+      .where(reports_table[:type].eq('ConfigReport')
+        .and(reports_table[:host_id].eq(host_status_table[:host_id])))
+      .exists
+  end
+
+  def no_reports
+    any_reports.not
+  end
+
+  def pending
+    status = host_status_table[:status] & 0b1111111111_0000000000_0000000000_0000000000_0000000000_0000000000
+
+    status.gteq(0b1_0000000000_0000000000_0000000000_0000000000_0000000000)
+      .and(status.lteq(0b1111111111_0000000000_0000000000_0000000000_0000000000_0000000000))
```

Comment:
Could you please write it in HEX? I believe it would be bit easier to read.

I'm not sure if masking with `&` works on the sql level, but if it does, then you need only `gt(0)`, as other segments have been masked out, if the mask (`&`) doesn't work, it's useless and you can remove it.